internal/renderers/femtovg/itemrenderer.rs

```diff
@@ -194,6 +195,77 @@ impl<'a, R: femtovg::Renderer + TextureImporter> GLItemRenderer<'a, R> {
     }
 }
 
+fn draw_glyphs<R: femtovg::Renderer + TextureImporter>(
+    layout: &sharedparley::Layout,
+    canvas: &mut Canvas<R>,
+    paint: &femtovg::Paint,
+    selection_foreground_color: Color,
+    selection_background_color: Color,
+) {
+    for line in layout.inner.lines() {
+        for item in line.items() {
+            match item {
+                parley::PositionedLayoutItem::GlyphRun(glyph_run) => {
+                    let run = glyph_run.run();
+
+                    let font_id =
+                        fonts::FONT_CACHE.with(|cache| cache.borrow_mut().font(run.font()));
+
+                    let brush = glyph_run.style().brush;
+
+                    let mut paint = paint.clone();
+                    if brush.is_selected {
+                        if let Some(start) = glyph_run.positioned_glyphs().next() {
+                            let selection_rect = PhysicalRect::new(
+                                PhysicalPoint::new(
+                                    start.x,
+                                    start.y + layout.y_offset - run.font_size(),
+                                ),
+                                PhysicalSize::new(
+                                    glyph_run.advance(),
+                                    run.font_size() + run.metrics().descent,
+                                ),
+                            );
+                            canvas.fill_path(
+                                &rect_to_path(selection_rect),
+                                &femtovg::Paint::color(to_femtovg_color(
+                                    &selection_background_color,
+                                )),
+                            );
+                        }
+
+                        paint.set_color(to_femtovg_color(&selection_foreground_color));
+                    }
+
+                    let glyphs = glyph_run.positioned_glyphs().map(|glyph: parley::Glyph| {
+                        femtovg::PositionedGlyph {
+                            x: glyph.x,
+                            y: glyph.y + layout.y_offset,
+                            font_id,
```

Comment:
As you pointed out, this is implemented slowly in femtovg. Given the single-font nature of glyph runs, I think the femtovg API should be called `stroke_glyph_run` and `fill_glyph_run`, take a font id as argument as well as a `PositionedGlyph` iterator that has the `font_id` field removed. If you'd like, I change my commits, but otherwise please feel free to make those changes in my branch in the femtovg fork. Let me know what you prefer or think :)

Reply:
Yes, let's move font_id out of PositionedGlyph. Additionally let's change the femtovg api to either use harfrust or yoke as @nicoburns suggested.

Reply:
Could you try https://github.com/slint-ui/femtovg/tree/simon/draw-glyph-run first? Move to harfrust makes sense (now that it's there), but I'd like to keep that separate from this PR.